HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/URLOpener.swift

```diff
@@ -22,11 +22,19 @@ public struct URLOpener {
     ///                 param is true if website was opened successfully
     ///                 param is false if opening failed
 
-    func openLink(url: URL, completion: ((Bool) -> Void)?) {
+    func openLink(url: URL, completion: (@MainActor @Sendable (Bool) -> Void)?) {
```

Comment:
[Clarification]: we support iOS 12 for the Gini Health SDK,
what will happen with the function itself `func openLink(url: URL, completion: (@MainActor @Sendable (Bool) -> Void)?) {`
it has `@MainActor @Sendable` parameter which is not available for iOS 12.
Will it compile?

Reply:
shall we add here `#if compiler(>=6.0)`?

Reply:
I've checkedn health sdk compiles in both Xcode 15 and 16. After a bit of research of why it does, I think @MainActor and @Sendable (including Sendable protocol) availability is bound to Swift/compiler version (Swift 5.5) rather than iOS version, though the MainActor (not the attribute) still seems to be bound to iOS version, MainActor.run on iOS 12 gives an error that it’s iOS 13+ only.

I think #if compiler(>=6.0) check could be added to get rid of Converting function value of type ‘@MainActor @Sendable (Bool) -> Void' to ‘(Bool) -> Void' loses global actor ‘MainActor'; this is an error in Swift 6 warning on line 27, but it doesn’t show up in Xcode 16, because in URLOpenerProtocol open url method there’s no @MainActor and @Sendable only if the compiler version is lower than 6.0, to comply with UIApplication.open url method definition, so there won’t be any error there in swift 6 mode, which is Xcode 16+ only. And UIApplication calls the completion block on the main thread anyways, as per documentation. @MainActor attribute just tells the compiler that the code will be executed on the main thread